core/src/main/scala/frameless/CatalystAbsolute.scala

```diff
@@ -0,0 +1,20 @@
+package frameless
+
+import scala.annotation.implicitNotFound
+
+/** Spark does not return always the same type on abs as the input was
+  */
+@implicitNotFound("Cannot compute absolute on type ${In}.")
+trait CatalystAbsolute[In, Out]
+
+object CatalystAbsolute {
+  private[this] val theInstance = new CatalystAbsolute[Any, Any] {}
+  private[this] def of[In, Out]: CatalystAbsolute[In, Out] = theInstance.asInstanceOf[CatalystAbsolute[In, Out]]
+
+  implicit val framelessAbsoluteBigDecimal: CatalystAbsolute[BigDecimal, java.math.BigDecimal]  = of[BigDecimal, java.math.BigDecimal]
+  implicit val framelessAbsoluteDouble    : CatalystAbsolute[Double, Double]                    = of[Double, Double]
+  implicit val framelessAbsoluteInt       : CatalystAbsolute[Int, Int]                          = of[Int, Int]
+  implicit val framelessAbsoluteLong      : CatalystAbsolute[Long, Long]                        = of[Long, Long]
+  implicit val framelessAbsoluteByte      : CatalystAbsolute[Byte, Byte]                        = of[Byte, Byte]
```

Comment:
Do we really need an absolute on Byte? I think someone can cast it to short before they do that. In a away, Byte is not a numeric type and we shouldn't tread it as such,

Reply:
Spark does support abs on byte though with the internal representation being tinyint. I am fine with removing byte from the abs-able types.